src/main/scala/com/cognite/sdk/scala/common/ReadableResource.scala

```diff
@@ -210,3 +214,35 @@ object RetrieveByExternalIdsWithIgnoreUnknownIds {
       value => value.items
     )
 }
+
+trait RetrieveByInstanceIds[R, F[_]] extends WithRequestSession[F] with BaseUrl {
+  def retrieveByInstanceIds(instanceIds: Seq[InstanceId]): F[Seq[R]]
+  def retrieveByInstanceId(instanceId: InstanceId): F[R] =
+    retrieveByInstanceIds(Seq(instanceId))
+      .flatMap(items =>
+        F.fromOption(
+          items.headOption,
+          SdkException("Unexpected empty response when retrieving item by InstanceId")
+        )
+      )
+}
+
+trait RetrieveByInstanceIdsWithIgnoreUnknownIds[R, F[_]] extends RetrieveByInstanceIds[R, F] {
+  override def retrieveByInstanceIds(ids: Seq[InstanceId]): F[Seq[R]] =
+    retrieveByInstanceIds(ids, ignoreUnknownIds = false)
+  def retrieveByInstanceIds(ids: Seq[InstanceId], ignoreUnknownIds: Boolean): F[Seq[R]]
+}
+
+object RetrieveByInstanceIdsWithIgnoreUnknownIds {
+  def retrieveByInstanceIds[F[_], R](
+      requestSession: RequestSession[F],
+      baseUrl: Uri,
+      cogniteIds: Seq[CogniteIdOrInstanceId],
+      ignoreUnknownIds: Boolean
+  )(implicit itemsDecoder: Decoder[Items[R]]): F[Seq[R]] =
+    requestSession.post[Seq[R], Items[R], ItemsWithIgnoreUnknownIds[CogniteIdOrInstanceId]](
```

Comment:
```suggestion
    requestSession.post[Seq[R], Items[R], ItemsWithIgnoreUnknownIds[InstanceId]](
```
still had a not on this method being called byInstanceIds but taking it CogniteIdOrInstanceId, should either be renamed or changed to accept according to name